rfcs/proposed/numa_support/increase-numa-support-availability.org

```diff
@@ -0,0 +1,115 @@
+# -*- fill-column: 80; -*-
+
+#+title: Improve predictability of API for NUMA support
+
+*Note:* This is a sub-RFC of the https://github.com/oneapi-src/oneTBB/pull/1535.
+Specifically, its section about "Increased availability of NUMA support".
+
+* Introduction
+oneTBB has a soft dependency on several variants of ~tbbbind~, which are loaded
+by the library as part of its initialization stage. In turn, each ~tbbbind~ has
+a hard dependency, i.e., relies on load-time linking, on a concrete version of
+the HWLOC library [1, 2]. The soft dependency of oneTBB on ~tbbbind~ allows the
+library to continue its execution even if the system loader is unable to resolve
+the hard dependency on HWLOC for ~tbbbind~. In this case, the HW topology is not
+discovered and the machine is seen as if all CPU cores were uniform, which is
+the default TBB behavior when NUMA constraints are not used. Thus, the following
+code returns meaningless values as these values are just ignored by oneTBB:
+
+#+begin_src C++
+std::vector<oneapi::tbb::numa_node_id> numa_nodes = oneapi::tbb::info::numa_nodes();
+std::vector<oneapi::tbb::core_type_id> core_types = oneapi::tbb::info::core_types();
+#+end_src
+
+An error is also not reported and the client code that uses NUMA support
+facilities may continue running expecting it to work as it was intended. Such
+behavior is not readily noticable by developers that use oneTBB and this
+represents the main problem with the current behavior.
+
+Having a dependency on a shared HWLOC library has a number of advantages:
+1. Code reuse with all of the positive consequences out of this. That's the
+   primary purpose of shared libraries.
+2. Sharing HWLOC context between its clients. This avoids performing the same
+   operations repeatedly with identical results.
+3. A drop-in replacement. Users are able to use their own version of HWLOC
+   without recompilation of oneTBB.
```

Comment:
My attitude to these "advantages" is that they are of little, if any, importance. No data supporting that "sharing HWLOC context" is provided, so it's kind of unproven. And I simply do not understand why somebody would want to "use their own version of HWLOC" or even what that "special" version would do.

However, importance of these features is not described, neither is a possible impact of their absence, Lack of this information may in general lead to suboptimal decisions.